stable_baselines3/common/utils.py

```diff
@@ -230,6 +230,22 @@ def check_for_correct_spaces(env: GymEnv, observation_space: spaces.Space, actio
         raise ValueError(f"Action spaces do not match: {action_space} != {env.action_space}")
 
 
+def check_shape_equal(space1: Union[spaces.Box, spaces.Dict], space2: Union[spaces.Box, spaces.Dict]) -> None:
```

Comment:
you should probably add a `keys` argument as the `norm_obs_keys` we have in `VecNormalize`, otherwise it will crash at load time, no?

Reply:
You mean, for example, if the subspaces are not all Box? I don't have any example of such env in mind but you're right. I would rather skip test for these case, see my next commit.

Reply:
on a second thought, I remember gym defines `shape` for all spaces (https://github.com/openai/gym/blob/v0.21.0/gym/spaces/space.py#L22) but I would double check that.

>  if the subspaces are not all Box? 

yes, a mixed space like one is an indicator (or a switch, do discrete) and the other one is the rest of the state (box).
that's also why we have the `norm_obs_keys` to allow skipping spaces we don't want to normalize or that can't be normalized.

Reply:
> on a second thought, I remember gym defines `shape` for all spaces (https://github.com/openai/gym/blob/v0.21.0/gym/spaces/space.py#L22) so we should be fine ,but I would double check that.

Correct! I didn't know.

It should work that way.